app/src/main/java/org/signal/cashu/service/DefaultCashuService.kt

```
@@ -0,0 +1,145 @@
+                transactionDao.insertTransaction(
+                    transaction.copy(
+                        amount = amount,
+                        status = TransactionStatus.COMPLETED
+                    )
+                )
+                return true
+            } else {
+                transactionDao.insertTransaction(transaction.copy(status = TransactionStatus.FAILED))
+                return false
+            }
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)